Helper/SiteAccess.php

```diff
@@ -0,0 +1,180 @@
+ */
+namespace EzSystems\RecommendationBundle\Helper;
+
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Symfony\SiteAccess as CurrentSiteAccess;
+use LogicException;
+
+class SiteAccess
+{
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    private $configResolver;
+
+    /** @var \eZ\Publish\Core\MVC\Symfony\SiteAccess */
+    private $siteAccess;
+
+    /** @var array $siteAccessConfig */
+    private $siteAccessConfig;
+
+    public function __construct(
+        ConfigResolverInterface $configResolver,
+        CurrentSiteAccess $siteAccess,
+        $siteAccessConfig
+    ) {
+        $this->configResolver = $configResolver;
+        $this->siteAccess = $siteAccess;
+        $this->siteAccessConfig = $siteAccessConfig;
+    }
+
+    /**
+     * Returns rootLocation by siteAccess name or by default siteAccess.
+     *
+     * @param string|null $siteAccessName
+     *
+     * @return int
+     */
+    public function getRootLocationBySiteAccessName($siteAccessName = null)
+    {
+        return $this->configResolver->getParameter(
+            'content.tree_root.location_id',
+            null,
+            $siteAccessName ?: $this->siteAccess->name
+        );
+    }
+
+    /**
+     * Returns list of rootLocations from siteAccess list.
+     *
+     * @param array $siteAccesses
+     *
+     * @return array
+     */
+    public function getRootLocationsBySiteAccesses(array $siteAccesses)
+    {
+        $rootLocations = [];
+
+        foreach ($siteAccesses as $siteAccess) {
+            $rootLocationId = $this->getRootLocationBySiteAccessName($siteAccess);
+
+            $rootLocations[$rootLocationId] = $rootLocationId;
+        }
+
+        return array_keys($rootLocations);
+    }
+
+    /**
+     * Returns languages based on mandatorId or siteaccess.
+     *
+     * @param null|int $mandatorId
+     * @param null|string $siteAccess
+     *
+     * @return array
+     */
+    public function getLanguages($mandatorId = null, $siteAccess = null)
+    {
+        if ($mandatorId) {
+            $languages = $this->getMainLanguagesBySiteAccesses(
+                $this->getSiteAccessesByMandatorId($mandatorId)
+            );
+        } elseif ($siteAccess) {
+            $languages = $this->configResolver->getParameter('languages', '', $siteAccess);
+        } else {
+            $languages = $this->configResolver->getParameter('languages');
+        }
+
+        if (empty($languages)) {
+            throw new LogicException(sprintf('No languages found using siteAccess or mandatorId'));
+        }
+
+        return $languages;
+    }
+
+    /**
+     * @param null|int $mandatorId
+     *
+     * @return array
+     *
+     * @throws NotFoundException
+     */
+    public function getSiteAccessesByMandatorId($mandatorId = 0)
+    {
+        if (0 == $mandatorId) {
```

Comment:
I assume that this line should looks like the follows:`if (!$mandatorId) {` or `if ($mandatorId === null) {`